### PR TITLE
Rustls swapped in for openssl

### PR DESCRIPTION
**Pull request by estk**
TODO:
- [x] Ensure Integration tests still pass
- [x] Setup CI
- [ ] Update docs

### CODE REVIEW

**Comment by estk** [comment]
Next step here is to setup CI so we can be sure security feature still works. In the mean time would you @ZaSkittles review this please.

File: examples/example-ssl.rs [hunk 1]
```diff
@@ -31,43 +29,46 @@ mod example {
...
-                builder
-                    .set_certificate_file(ccert, X509_FILETYPE_PEM)
-                    .unwrap();
-                builder
-                    .set_private_key_file(ckey, X509_FILETYPE_PEM)
-                    .unwrap();
-                builder.check_private_key().unwrap();
+                let certs = {
+                    let file = fs::File::open(ccert).expect("cannot open private key file");
+                    let mut reader = BufReader::new(file);
```
Comment (ZaSkittles) [inline examples/example-ssl.rs:41]: is there a special security centric reader or anything. Mostly some special way to mark that this memory contains privileged data or info? 
Might be overkill but I honestly don't know.

File: examples/example-ssl.rs [hunk 2]
```diff
@@ -31,43 +29,46 @@ mod example {
...
+                };
+
+                let keys = {
+                    let file = fs::File::open(ckey).expect("cannot open private key file");
+                    let mut reader = BufReader::new(file);
+                    rustls::internal::pemfile::rsa_private_keys(&mut reader)
+                        .expect("unable to read key file")
+                };
+
+                let key = keys[0].clone();
```
Comment (ZaSkittles) [inline examples/example-ssl.rs:53]: bounds check to make sure it's non empty? Oh I guess this is also just example code tho

File: src/client/network.rs [hunk 3]
```diff
@@ -21,38 +22,32 @@ use crate::error::Result;
...
+        SecurityConfig { rustls_config }
     }
 
     /// Initiates a client-side TLS session with/without performing hostname verification.
-    pub fn with_hostname_verification(self, verify_hostname: bool) -> SecurityConfig {
-        SecurityConfig {
-            verify_hostname,
-            ..self
-        }
+    #[cfg(feature = "dangerous_configuration")]
```
Comment (ZaSkittles) [inline src/client/network.rs:39]: I like the dangerous config usage/aspect here.

**Comment by ZaSkittles** [comment]
Nothing too major, a couple of little questions. But looks reasonable and cleaner than before.

**Review by ZaSkittles — APPROVED** [review]

File: examples/example-ssl.rs (hunk 1 above)
Comment (estk) [reply examples/example-ssl.rs:41]: Huh, never thought about it, do you have an example from another lang?

File: examples/example-ssl.rs (hunk 2 above)
Comment (estk) [reply examples/example-ssl.rs:53]: Lifted from the crate's examples here: https://github.com/ctz/rustls/blob/master/rustls-mio/examples/tlsclient.rs#L412
Maybe I'll add a comment linking this..

**Pull request merged**